app/workers/company_video_worker.rb

```diff
@@ -0,0 +1,31 @@
+class CompanyVideoWorker
+  include Sidekiq::Worker
+
+  def perform(company_id)
+    company = Company.find(company_id)
+    if company.youtube?
+      channel_id = company.youtube.split('channel/').last
+      channel = authenticate(channel_id)
```

Comment:
what if any exception? for example `authenticate` returns some (incorrect credentials, or timeout, etc).

Reply:
or for example channel not found, etc